plugins/helpers.js

```diff
@@ -1,5 +1,12 @@
+const state = new WeakMap();
+
 function forwardEvents(src, dst, ...events) {
-    events.forEach((name) => src.on(name, dst.emit.bind(dst)));
+    events.forEach((name) => src.on(name, (...args) => {
+        if (state.has(dst) && state.get(dst).includes(name)) return;
+        state.set(src, (state.get(src) || []).concat(name));
+        dst.emit(name, ...args);
+        state.set(src, state.get(src).filter(_name => _name !== name));
```

Comment:
Sorry I didn't follow here, why you need to add the event to the state and then remove it after emit?

Reply:
This was to prevent recursive emits to itself (it happens when applying this function in the opposite way), and it worked for synchronous event listeners because of the synchronous nature of event emitters.
But for this to be complete, I realized I had to cover the asynchronous cases. I'm pushing another change as an attempt to address this.